entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/extension/KotlinTypeExt.kt

```diff
@@ -18,19 +17,8 @@ fun KotlinType.isResource(): Boolean {
         .any { it == "godot.Resource" }
 }
 
-fun KotlinType.isVariantArray(): Boolean {
-    return this.getJetTypeFqName(false) == "godot.core.VariantArray"
-        || this
-        .supertypes()
-        .map { it.getJetTypeFqName(false) }
-        .any { it == "godot.core.VariantArray" }
-}
```

Comment:
Why don't we support it anymore to extend the variant array class with your own implementation?

Reply:
Hmm, why would you do that?

Reply:
provide additional functionality to variantArray. Setup some kind of type hierarchy with it and do stuff depending on the type of it. I don't know. There probably is a use case and i don't see why we can't or shouldn't support it now.

Reply:
for example implement functions in a custom variant array.

Custom VariantArray class Foo has function splitByType()
Custom VariantArray class Bar has function splitByName()

This can't be done through extension functions as one function should be only applicable to a specific instance of a custom variantArray

Reply:
I'm not sure that we can support that as `Variant` needs to be aware of that custom type. 

```kotlin
@RegisterProperty
var myCustomArray = MyCustomArray()
```

So when godot sets this property, it passes on a `Variant` which we call `unwrap` on it will return a `VariantArray`. In order to support that we need give that information to variant at runtime.

Also supporting it also means users can add additional properties to it: `class MyCustomArray(val data: ...): VariantArray<Int>` which are additional information we need to keep track at runtime.

I'm not convince the user value it adds outweighs the complexity needed to support it.

Extension function is a more idiomatic way to implement this use case (https://medium.com/@elizarov/extension-oriented-design-13f4f27deaee)

```kotlin
fun VariantArray<SomeType>.doSomethingWithSomeType() = ...
fun VariantArray<Int>.sum() = ...
```

Reply:
Also, I think we should really think of core types as final classes. They can't be inherited from GDScript as they are akin to primitives.

Reply:
Convinced

Reply:
@raniejade do you look with @piiertho regarding finalizing the core classes? I find that a good idea :-)